### PR TITLE
[15.0][MIG] edi_voxel_sale_order_import: Migration to version 15.0

### PR DESCRIPTION
**Pull request by sergio-teruel**
cc @Tecnativa TT36456

ping @carlosdauden @CarlosRoca13 

depends on:
- [x] https://github.com/OCA/edi/pull/676

### CODE REVIEW

**Review by carlosdauden — APPROVED** [review]
Migration commit LGTM

**Comment by sergio-teruel** [comment]
Rebase done

File: edi_voxel_sale_order_import_oca/models/sale_order.py [hunk 1]
```diff
@@ -0,0 +1,384 @@
...
+            vals.update(note="\n\n".join(comments))
+
+    def _parse_product_list_data_voxel(self, order, xml_root, error_msgs):
+        line_elements = xml_root.xpath("//ProductList/Product")
+        so_line_obj = self.env["sale.order.line"]
+        for line_element in line_elements:
+            line_vals = {"order_id": order.id}
+            self._parse_product_voxel(line_vals, line_element)
+            self._parse_qty_uom_voxel(line_vals, line_element)
+            line_vals = so_line_obj.play_onchanges(line_vals, list(line_vals)[1:])
```
Comment (ernestotejeda) [inline edi_voxel_sale_order_import_oca/models/sale_order.py]: Include the first value (order_id) to take into account related values like company_id when applying changes.

```suggestion
            line_vals = so_line_obj.play_onchanges(line_vals, list(line_vals))
```

**Review by ernestotejeda — CHANGES_REQUESTED** [review]

File: edi_voxel_sale_order_import_oca/models/sale_order.py (hunk 1 above)
Comment (sergio-teruel) [reply edi_voxel_sale_order_import_oca/models/sale_order.py]: Changes done!!

**Comment by pedrobaeza** [comment]
@sergio-teruel as this depends on `onchange_helper` that is Beta, it can't be `Production/Stable`. Please change status or promote onchange_helper to Stable.

**Comment by pedrobaeza** [comment]
@simahawk why a module with a name `pdf_helper` is here in OCA/edi? Is doesn't make too much sense. Anyway, their tests are failing, as `TreeCase` doesn't exist. Please fix it, as it's blocking this PR.

**Comment by pedrobaeza** [comment]
Please check CI

**Comment by simahawk** [comment]
> @simahawk why a module with a name `pdf_helper` is here in OCA/edi? Is doesn't make too much sense. Anyway, their tests are failing, as `TreeCase` doesn't exist. Please fix it, as it's blocking this PR.

It has been ripped out of business document import in 14 a long time ago and I didn't want to move it to minimize the impact on everybody else. I'm not against moving it (maybe to server-tools).

**Comment by pedrobaeza** [comment]
Well, that one is making the CI to fail. Please check at least that. And with pip, it's supposed that the OCA repo where it's hosted is not important.

**Comment by simahawk** [comment]
> Well, that one is making the CI to fail. Please check at least that. And with pip, it's supposed that the OCA repo where it's hosted is not important.

Not for CI, or for me. Is for the ones not relying on it :wink: 

Et voilá https://github.com/OCA/edi/pull/811

**Comment by carlosdauden** [comment]
@sergio-teruel rebase to test

**Comment by sergio-teruel** [comment]
Done!

**Review by pedrobaeza — APPROVED** [review]
/ocabot merge nobump

**Pull request merged**